src/image-viewer/hooks/useRotate.ts

```diff
@@ -1,14 +1,31 @@
 // 旋转控制
-import { useCallback, useState } from 'react';
+import { useCallback, useRef, useState } from 'react';
 
 const useRotate = () => {
+  // There is an useEffect in the line 472 of ImageViewerModal.tsx, so we need to use a ref to store the rotateZ value.
+  const rotRef = useRef(0);
   const [rotateZ, setRotateZ] = useState(0);
 
   const onRotate = useCallback((ROTATE_COUNT: number) => {
-    setRotateZ((rotateZ) => rotateZ + ROTATE_COUNT);
+    setRotateZ((rotateZ) => {
+      rotRef.current = rotateZ + ROTATE_COUNT;
+      return rotateZ + ROTATE_COUNT;
+    });
   }, []);
 
-  const onResetRotate = useCallback(() => setRotateZ(0), []);
+  const onResetRotate = useCallback(() => {
+    let degreeToRotate = rotRef.current % 360;
```

Comment:
Maybe we should take the remainder of 360 every time we rotate, instead of calculating it when resetting？